Codegenerator fails on Windows

Code doens't generate however i can't find a error in my JSON. Any idea what im doing wrong?

```
Exception in thread "main" java.lang.RuntimeException: Could not generate api file for 'App'
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:296)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:188)
        at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
Caused by: java.lang.IllegalArgumentException: character to be escaped is missing
        at java.util.regex.Matcher.appendReplacement(Unknown Source)
        at java.util.regex.Matcher.replaceAll(Unknown Source)
        at java.lang.String.replaceAll(Unknown Source)
        at io.swagger.codegen.languages.PhpClientCodegen.toPackagePath(PhpClientCodegen.java:124)
        at io.swagger.codegen.languages.PhpClientCodegen.apiFileFolder(PhpClientCodegen.java:212)
        at io.swagger.codegen.DefaultCodegen.apiFilename(DefaultCodegen.java:1612)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:275)
        ... 2 more
```

Im now using a fresh clone from master. (swagger-generator does fail on building though. Don't know if this is a problem?) I can generate all languages except for PHP strange enough. It somehow can't create files for PHP. I tried using cygwin and windows terminal.

```
Exception in thread "main" java.lang.RuntimeException: Could not generate model 'inline_response_200_11'
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:215)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:188)
        at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
Caused by: java.lang.IllegalArgumentException: character to be escaped is missing
        at java.util.regex.Matcher.appendReplacement(Unknown Source)
        at java.util.regex.Matcher.replaceAll(Unknown Source)
        at java.lang.String.replaceAll(Unknown Source)
        at io.swagger.codegen.languages.PhpClientCodegen.toPackagePath(PhpClientCodegen.java:132)
        at io.swagger.codegen.languages.PhpClientCodegen.modelFileFolder(PhpClientCodegen.java:224)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:196)
        ... 2 more
```

--- a/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java
+++ b/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java
@@ -19,7 +19,12 @@ import java.util.regex.Matcher;
 
 import org.apache.commons.lang3.StringUtils;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
+    static Logger LOGGER = LoggerFactory.getLogger(PhpClientCodegen.class);
+
     public static final String VARIABLE_NAMING_CONVENTION = "variableNamingConvention";
     public static final String PACKAGE_PATH = "packagePath";
     public static final String SRC_BASE_PATH = "srcBasePath";
@@ -109,6 +114,8 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
     }
 
     public String toPackagePath(String packageName, String basePath) {
+        LOGGER.info("============================");
+        LOGGER.info("packageName = " + packageName);
         packageName = packageName.replace(invokerPackage, "");
         if (basePath != null && basePath.length() > 0) {
             basePath = basePath.replaceAll("[\\\\/]?$", "") + File.separatorChar;
@@ -120,6 +127,7 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
         } else { // for windows
             regFirstPathSeparator = "^\\\\";
         }
+        LOGGER.info("regFirstPathSeparator = " + regFirstPathSeparator);
 
         String regLastPathSeparator;
         if ("/".equals(File.separator)) { // for mac, linux
@@ -127,6 +135,8 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
         } else { // for windows
             regLastPathSeparator = "\\\\$";
         }
+        LOGGER.info("regLastPathSeparator= " + regLastPathSeparator);
+        LOGGER.info("packageName = " + packageName);
 
         return (getPackagePath() + File.separatorChar + basePath
                     // Replace period, backslash, forward slash with file separator in package name
@@ -218,11 +228,11 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
 
     @Override
     public String apiFileFolder() {
-        return (outputFolder + "/" + toPackagePath(apiPackage(), srcBasePath));
+        return (outputFolder + "/" + toPackagePath(apiPackage, srcBasePath));
     }
 
     public String modelFileFolder() {
-        return (outputFolder + "/" + toPackagePath(modelPackage(), srcBasePath));
+        return (outputFolder + "/" + toPackagePath(modelPackage, srcBasePath));
     }
 
     @Override

--- a/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java
+++ b/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java
@@ -15,6 +15,7 @@ import java.io.File;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.regex.Matcher;
 
 import org.apache.commons.lang3.StringUtils;
 
@@ -129,7 +130,7 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
 
         return (getPackagePath() + File.separatorChar + basePath
                     // Replace period, backslash, forward slash with file separator in package name
-                    + packageName.replaceAll("[\\.\\\\/]", File.separator)
+                    + packageName.replaceAll("[\\.\\\\/]", Matcher.quoteReplacement(File.separator))
                     // Trim prefix file separators from package path
                     .replaceAll(regFirstPathSeparator, ""))
                     // Trim trailing file separators from the overall path

--- a/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java
+++ b/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java
@@ -114,8 +114,6 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
     }
 
     public String toPackagePath(String packageName, String basePath) {
-        LOGGER.info("============================");
-        LOGGER.info("packageName = " + packageName);
         packageName = packageName.replace(invokerPackage, "");
         if (basePath != null && basePath.length() > 0) {
             basePath = basePath.replaceAll("[\\\\/]?$", "") + File.separatorChar;
@@ -127,7 +125,6 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
         } else { // for windows
             regFirstPathSeparator = "^\\\\";
         }
-        LOGGER.info("regFirstPathSeparator = " + regFirstPathSeparator);
 
         String regLastPathSeparator;
         if ("/".equals(File.separator)) { // for mac, linux
@@ -135,8 +132,6 @@ public class PhpClientCodegen extends DefaultCodegen implements CodegenConfig {
         } else { // for windows
             regLastPathSeparator = "\\\\$";
         }
-        LOGGER.info("regLastPathSeparator= " + regLastPathSeparator);
-        LOGGER.info("packageName = " + packageName);
 
         return (getPackagePath() + File.separatorChar + basePath
                     // Replace period, backslash, forward slash with file separator in package name

